auv_sim/auv_sim_bridge/include/auv_sim_bridge/simulation_mock_ros.h

```diff
@@ -40,12 +42,49 @@ class SimulationMockROS {
   double standard_pressure_;  /// kPa
   double battery_voltage_;    /// V
   double battery_current_;    /// A
+  bool dvl_enabled_;
+  ros::Rate dvl_rate_;  /// Hz
+  double latest_altitude_;
+  Eigen::Matrix3d linear_covariance_;
+  Eigen::Matrix3d noise_transform_;
 
-  bool setDVLPing(std_srvs::SetBoolRequest &req,
-                  std_srvs::SetBoolResponse &resp) {
-    // this is a dummy service, does nothing
+  geometry_msgs::Twist rotateVelocity(
+      const geometry_msgs::Twist &input_velocity, double angle) {
+    const double theta = angle * M_PI / 180.0;
+    Eigen::Matrix3d rotation_matrix;
+    rotation_matrix << std::cos(theta), -std::sin(theta), 0, std::sin(theta),
+        std::cos(theta), 0, 0, 0, 1;
+
+    Eigen::Vector3d velocity_vector(input_velocity.linear.x,
+                                    input_velocity.linear.y,
+                                    input_velocity.linear.z);
+
+    Eigen::Vector3d rotated_velocity = rotation_matrix * velocity_vector;
+
+    geometry_msgs::Twist velocity_msg;
+    velocity_msg.linear.x = rotated_velocity.x();
+    velocity_msg.linear.y = rotated_velocity.y();
+    velocity_msg.linear.z = rotated_velocity.z();
+    return velocity_msg;
+  }
+
+  geometry_msgs::Twist addNoiseToTwist(const geometry_msgs::Twist &input) {
```

Comment:
Instead of making a copy, you could write to the given argument in place by removing const and leaving the ref (&). But I don't even think it will matter that much

Reply:
used reference instead of making a copy as you suggested